### PR TITLE
Include .config files in AllowedReferenceRelatedFileExtensions

### LINKED ISSUE #1307: Include .config file when copying output files of a dependent project
Consider scenario where project A references project B and project B has an App.config file. The build copies .dll/.exe and .pdb files of B to the output directory of A, but doesn't currently copy .dll.config/.exe.config files that may contain binding redirects and other configuration.

I propose the build copies the configuration file as well, so that binding redirects are not lost. Since the latest CoreFX binaries heavily rely on presence of binding redirects having binding redirects flow seamlessly thru build system is becoming more and more important.

### PR DESCRIPTION
**Pull request by jeffkl**
Closes #1307

### CODE REVIEW

**Comment by jeffkl** [comment]
@dotnet-bot test Windows_NT Build for Desktop please

File: src/XMakeTasks/Microsoft.Common.CurrentVersion.targets [hunk 1]
```diff
@@ -576,7 +576,8 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <AllowedReferenceRelatedFileExtensions Condition=" '$(AllowedReferenceRelatedFileExtensions)' == '' ">
       .pdb;
       .xml;
-      .pri
+      .pri;
+      .config
```
Comment (rainersigwald) [inline src/XMakeTasks/Microsoft.Common.CurrentVersion.targets]: I think it'd be better to explicitly have `.dll.config` and `.exe.config`.

Comment (AndyGerlicher) [inline src/XMakeTasks/Microsoft.Common.CurrentVersion.targets]: Change to `.exe.config` and `.dll.config`

Comment (jeffkl) [inline src/XMakeTasks/Microsoft.Common.CurrentVersion.targets]: Should I just do `.dll.config`?  Isn't `exe.config` already copied?

**Comment by jeffkl** [comment]
Nevermind, after testing this it only worked when I used `.dll.config` and `.exe.config`.

**Pull request merged**